pkg/controllers/vrep/replication_reconciler.go

```diff
@@ -238,33 +237,65 @@ func (r *ReplicationReconciler) checkSandboxExists() error {
 	return nil
 }
 
-// choose the source host and target host
-// (first host where db is up in the specified cluster)
+// filters services based on svcName if not empty and main/sandbox cluster
+// picks first service that matches. if svcName is empty this will be the first service assigned to the database
+// if svcName is non-empty it will be the service matching that name
+// returns a valid hostname for the service or an error if a valid matching service couldn't be found
+func (r *ReplicationReconciler) getServiceHostname(vdb *vapi.VerticaDB,
+	sandboxName, svcName string) (ip string, err error) {
+	clusterName := vdb.Name
+	if sandboxName != vapi.MainCluster {
+		clusterName += " (sandbox " + sandboxName + ")"
+	}
+
+	if svcName != "" {
+		inputName := types.NamespacedName{Namespace: vdb.Namespace, Name: svcName}
+		for i := range vdb.Spec.Subclusters {
+			sc := &vdb.Spec.Subclusters[i]
+			scSvcName := names.GenExtSvcName(vdb, sc)
+			if inputName == scSvcName && sandboxName == vdb.GetSubclusterSandboxName(sc.Name) {
+				return fmt.Sprintf("%s.%s.svc.cluster.local", scSvcName.Name, scSvcName.Namespace), nil
```

Comment:
For precaution try to retrieve the service.

Reply:
you mean the way that i did previously?